soundata/core.py

```diff
@@ -361,6 +377,459 @@ def validate(self, verbose=True):
         )
         return missing_files, invalid_checksums
 
+    @lru_cache(maxsize=None)  # Setting maxsize to None for an unbounded cache
+    def compute_clip_statistics(self):
+        durations = [
+            len(self.clip(c_id).audio[0]) / self.clip(c_id).audio[1]
```

Comment:
Should we add a line checking if the attribute `audio` is present in the dataset just in case?